tftypes/value_test.go

```diff
@@ -1721,6 +1898,59 @@ func TestValueString(t *testing.T) {
 			in:       Value{},
 			expected: "invalid typeless tftypes.Value<>",
 		},
+		"unknown-bool": {
+			in:       NewValue(Bool, UnknownValue),
+			expected: "tftypes.Bool<unknown>",
+		},
+		"unknown-bool-with-nullness-refinement": {
+			in: NewValue(Bool, UnknownValue).Refine(refinement.Refinements{
+				refinement.KeyNullness: refinement.NewNullness(false),
+			}),
+			expected: `tftypes.Bool<unknown, not null>`,
+		},
+		"unknown-string": {
+			in:       NewValue(String, UnknownValue),
+			expected: "tftypes.String<unknown>",
+		},
+		"unknown-string-with-multiple-refinements": {
+			in: NewValue(String, UnknownValue).Refine(refinement.Refinements{
+				refinement.KeyNullness:     refinement.NewNullness(false),
+				refinement.KeyStringPrefix: refinement.NewStringPrefix("str://"),
+			}),
+			expected: `tftypes.String<unknown, not null, prefix = "str://">`,
+		},
+		"unknown-number": {
+			in:       NewValue(Number, UnknownValue),
+			expected: "tftypes.Number<unknown>",
+		},
+		"unknown-number-with-multiple-refinements": {
+			in: NewValue(Number, UnknownValue).Refine(refinement.Refinements{
+				refinement.KeyNullness:         refinement.NewNullness(false),
+				refinement.KeyNumberLowerBound: refinement.NewNumberLowerBound(big.NewFloat(5), true),
+				refinement.KeyNumberUpperBound: refinement.NewNumberUpperBound(big.NewFloat(10), true),
+			}),
+			expected: `tftypes.Number<unknown, not null, lower bound = 5 (inclusive), upper bound = 10 (inclusive)>`,
+		},
+		"unknown-number-float-with-multiple-refinements": {
+			in: NewValue(Number, UnknownValue).Refine(refinement.Refinements{
+				refinement.KeyNullness:         refinement.NewNullness(false),
+				refinement.KeyNumberLowerBound: refinement.NewNumberLowerBound(big.NewFloat(5.67), true),
+				refinement.KeyNumberUpperBound: refinement.NewNumberUpperBound(big.NewFloat(10.123), true),
+			}),
+			expected: `tftypes.Number<unknown, not null, lower bound = 5.67 (inclusive), upper bound = 10.123 (inclusive)>`,
+		},
+		"unknown-list": {
+			in:       NewValue(List{ElementType: String}, UnknownValue),
+			expected: "tftypes.List[tftypes.String]<unknown>",
+		},
+		"unknown-list-with-multiple-refinements": {
+			in: NewValue(List{ElementType: String}, UnknownValue).Refine(refinement.Refinements{
+				refinement.KeyNullness:                   refinement.NewNullness(false),
+				refinement.KeyCollectionLengthLowerBound: refinement.NewCollectionLengthLowerBound(2),
+				refinement.KeyCollectionLengthUpperBound: refinement.NewCollectionLengthUpperBound(7),
+			}),
+			expected: `tftypes.List[tftypes.String]<unknown, not null, length lower bound = 2, length upper bound = 7>`,
+		},
```

Comment:
We can bikeshed the display of these attributes since there is no established pattern ATM (core displays the `cty` methods when showing data consistency errors, but doesn't have a "user friendly" version of the data defined). Other thoughts:

```
tftypes.String<unknown, refinements = [not null, prefix = "str://"]>
```

```
tftypes.String<unknown, will not be null, will have prefix "str://">
```